channels/chan_pjsip.c

```diff
@@ -1564,13 +1572,18 @@ static int send_topology_change_refresh(void *data)
 {
 	struct topology_change_refresh_data *refresh_data = data;
 	struct ast_sip_session *session = refresh_data->session;
+	enum ast_channel_state state = ast_channel_state(session->channel);
+	enum ast_sip_session_refresh_method method = AST_SIP_SESSION_REFRESH_METHOD_INVITE;
 	int ret;
 	SCOPE_ENTER(3, "%s: %s\n", ast_sip_session_get_name(session),
 		ast_str_tmp(256, ast_stream_topology_to_str(refresh_data->media_state->topology, &STR_TMP)));
 
+	if (session->early_confirmed && (state == AST_STATE_RINGING || state == AST_STATE_RING)) {
```

Comment:
Add a comment specifying RFC 6337, as well as why ringing and ring.

Reply:
Done. Also, now the `early_confirmed` flag is also set whenever we receive a reliable 1xx response with SDP, which is how it should be.